portafly/src/components/shared/data-list/DataListContext.tsx

```diff
@@ -14,17 +14,23 @@ import {
   useBulkActions,
   defaultBulkActions,
   BulkActionsState,
-  bulkActionsReducer
+  bulkActionsReducer,
+  useData,
+  DataState,
+  dataReducer
 } from 'components/shared/data-list/reducers'
 
 export type State = {
+  data: DataState
   table: TableState
   filters: FiltersState
   pagination: PaginationState
   bulkActions: BulkActionsState
 }
 
 const defaultState: State = {
+  data: [],
+  // TODO: We won't need to store this when PF updates the way they craft tables
```

Comment:
I'm not so sure of this after all. The logic behind our table data is pretty heavy (rendering components, translations, predicates..) and it could have a great impact in performance, due to the high amount of re-renders a Table has.

Reply:
I'll remove the comment. We'll need to give a second look to the table, and aim to remove those rows and columns (which store components) from the state.